.github/workflows/ci.yaml

```diff
@@ -0,0 +1,176 @@
+name: CI
+
+on:
+  push:
+    branches:
+      - "**"
+    tags:
+      - "[0-9]+.[0-9]+.[0-9]+**"
+
+env:
+  GO_VERSION: 1.16
+  GOFLAGS: -mod=readonly
+
+jobs:
+  # Note: currently jobs.<job-id>.(runs-on|strategy) cannot use global '${{ env }}'.
+  #
+  # Note: currently ${{ env }} cannot define dynamically initialized values.
+  environment:
+    name: Configure environment
+    runs-on: ubuntu-latest # TODO: until static global env can be used through ${{ env }}.
+    outputs:
+      git_refname: ${{ steps.set-git-refname.outputs.git_refname }}
+      github_runner_default_os: ubuntu-latest
+
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v2
+
+      - name: Set Git refname
+        id: set-git-refname
+        run: echo ::set-output name=git_refname::$(echo "${{ github.ref }}" | sed -r 's@refs/(heads|pull|tags)/@@g' )
+
+  static-checks:
+    name: Run static code analysis
+    needs:
+      - environment
+    runs-on: ${{ needs.environment.outputs.github_runner_default_os }}
+
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v2
+
+      - name: Setup Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: ${{ env.GO_VERSION }}
+
+      - name: Cache licenses
+        id: cache-licenses
+        uses: actions/cache@v2
+        with:
+          path: .licensei.cache
+          key: licensei-v1-${{ needs.environment.outputs.git_refname }}-${{ hashFiles('go.sum') }}
+          restore-keys: |
+            - licensei-v1-${{ needs.environment.outputs.git_refname }}
+            - licensei-v1-master
+            - licensei-v1
+
+      - name: Download license information for dependencies
+        env:
+          GITHUB_TOKEN: ${{ github.token }} # Note: this is required for licensei authentication to avoid rate-limiting.
+        run: make license-cache
+
+      - name: List license information for dependencies
+        run: ./bin/licensei list
+
+      - name: Check dependency licenses
+        run: make license-check
+
+      - name: Run linter
```

Comment:
This could also make use of the dependency cache as many linters may require dependencies to be present.

Reply:
Fixed in 953cf26.